tel2puml/otel_to_pv/config.py

```diff
@@ -26,19 +30,264 @@ class SpanSpec(TypedDict):
+            key_values, key_paths
+        ):
+            if len(priority_key_values) != len(priority_key_paths):
+                raise ValueError(
+                    "Input Priority key values and priority key paths must "
+                    "be interables of the same length, single strings or None,"
+                    " or a combination of the two that is consitent in terms "
+                    "of length"
+                )
+
+    @staticmethod
+    def validate_value_paths_with_key_values(
+        value_paths: Sequence[tuple[str | None, ...]],
+        key_values: Sequence[tuple[str | None, ...]],
+    ) -> None:
+        """Validates value paths with key values.
+
+        :param value_paths: The value paths
+        :type value_paths: :class:`Sequence`[`tuple`[`str` |
+        :class:`None`, ...]]
+        :param key_values: The key values
+        :type key_values: :class:`Sequence`[`tuple`[`str` |
+        :class:`None`, ...]]
+        """
+        if len(value_paths) != len(key_values):
+            raise ValueError(
+                "Input value paths and input key values must have the same "
+                "length"
+            )
+        for priority_value_paths, priority_key_values in zip(
+            value_paths, key_values
+        ):
+            if len(priority_key_values) != len(priority_value_paths):
+                raise ValueError(
+                    "Input priority value paths and priority key values must "
+                    "be interables of the same length, single strings or None,"
+                    " or a combination of the two that is consitent in terms "
+                    "of length"
+                )
+            for value_path, key_value in zip(
+                priority_value_paths, priority_key_values
+            ):
+                if value_path is None and key_value is not None:
+                    raise ValueError(
+                        "If a key value is provided, a value path must also be"
+                        " provided"
+                    )
+
+    @staticmethod
+    def field_spec_key_path_to_jq_key_path(
+        key_path: str | Iterable[str],
+    ) -> tuple[str, ...]:
+        """Converts field spec key path to jq key path.
+
+        :param key_path: The key path
+        :type key_path: `str` | `Iterable`[`str`]
+        :return: The jq key path
+        :rtype: `tuple`[`str`, ...]
+        """
+        if isinstance(key_path, str):
+            return (key_path,)
+        try:
+            priority_key_path = tuple(iter(key_path))
+            for key in priority_key_path:
+                if not isinstance(key, str):
+                    raise ValueError(
+                        "Priority key path, within an iterable, must be a "
+                        "string"
+                    )
+            return priority_key_path
+        except TypeError:
+            raise ValueError("Key path must be iterable or a string")
```

Comment:
Do we want to raise a TypeError here?

Reply:
Yeah thats better idea actually - updated